src/app/components/news/news.component.ts

```diff
@@ -1,25 +1,19 @@
-import { Component, OnInit, Inject, Input } from '@angular/core';
-import { NewsService, ActualNewsService } from '../../services/news.service';
+import { Component, OnInit, Inject } from '@angular/core';
+import { NewsService } from '../../services/news.service';
 import { News } from '../../models/News';
 
 @Component({
   selector: 'app-news',
   templateUrl: './news.component.html',
   styleUrls: ['./news.component.scss'],
-  // Here we choose which news service is used
-  providers: [{ provide: 'NewsService', useClass: ActualNewsService }]
 })
 export class NewsComponent implements OnInit {
   news: News[];
 
-  constructor(@Inject('NewsService') newsService: NewsService) {
-    newsService.getNews().subscribe(
-      news => {
-        this.news = news;
-      });
-  }
+  constructor(@Inject('NewsService') private newsService: NewsService) {}
 
   ngOnInit(): void {
+    this.newsService.getNewsObserver().subscribe(news => { this.news = news; });
```

Comment:
Probably worth double checking, but you might need to explicitly handle unsubscribing from your own observables/subjects in `ngOnDestroy `to avoid memory leaks. I don't think any kind of automatic cleanup exists?